examples/dark-mode/app/root.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  Links,
+  LiveReload,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  useLoaderData
+} from "remix";
+import type { LoaderFunction } from "remix";
+
+import {
+  NonFlashOfWrongThemeEls,
+  ThemeProvider,
+  useTheme
+} from "~/utils/theme-provider";
+import type { Theme } from "~/utils/theme-provider";
+import { getThemeSession } from "./utils/theme.server";
+
+export type LoaderData = {
+  theme: Theme | null;
+};
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const themeSession = await getThemeSession(request);
+
+  const data: LoaderData = {
+    theme: themeSession.getTheme()
+  };
+
+  return data;
+};
+
+export function App() {
```

Comment:
Why is this being exported? Also the code is simple enough to combine it into a single component with the default export

Reply:
This was a typo, I've removed the export. I think it still needs to be a separate component though because `useTheme` is basically a wrapper around `useContext` so we need the provider in a parent component?